bundles/org.openhab.binding.haassohnpelletoven/src/main/java/org/openhab/binding/haassohnpelletoven/internal/HaassohnpelletstoveHandler.java

```diff
@@ -0,0 +1,390 @@
+                statusDescr = "Parameter 'refresh Rate' greater then 0 and less then 1000.";
+                validConfig = false;
+            }
+
+            if (config.hostIP == null) {
+                errors += " Parameter 'hostIP' must be configured.";
+                statusDescr = "IP Address must be configured!";
+                validConfig = false;
+            } else if (!new IpAddressValidator().isValid(config.hostIP)) {
+                errors += " 'hostIP' is no valid IP address.";
+                logger.debug("{} is no valid IP address.", config.hostIP);
+                statusDescr = "No valid IP-Adress configured.";
+                validConfig = false;
+            }
+
+            if (config.hostPIN == null) {
+                errors += " Parameter 'hostPin' must be configured.";
+                statusDescr = "PIN must be configured!";
+                validConfig = false;
+            } else if (!new PinValidator().isValid(config.hostPIN)) {
+                errors += " 'hostPIN' is no valid PIN. PIN consists of 4-digit numbers.";
+                logger.debug("{} is no valid PIN. PIN consists of 4-digit numbers.", config.hostPIN);
+                statusDescr = "No valid PIN configure. A valid PIN consists of 4-digit numbers.";
+                validConfig = false;
+            }
+            errors = errors.trim();
+        }
+
+        Helper message = new Helper();
+        message.setStatusDescription(statusDescr);
+        if (validConfig) {
+            serviceCommunication.setConfig(config);
+            if (serviceCommunication.refreshOvenConnection(message, this.getThing().getUID().toString())) {
+                if (updateOvenData(null)) {
+                    updateStatus(ThingStatus.ONLINE);
+                    updateLinkedChannels();
+                }
+            } else {
+
+                logger.debug("Setting thing '{}' to OFFLINE: {}", getThing().getUID(), errors);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+            }
+
+        } else {
+            logger.debug("Setting thing '{}' to OFFLINE: {}", getThing().getUID(), errors);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+        }
+    }
+
+    private void updateLinkedChannels() {
+
+        verifyLinkedChannel(CHANNEL_isTemp);
+        verifyLinkedChannel(CHANNEL_mode);
+        verifyLinkedChannel(CHANNEL_prg);
+        verifyLinkedChannel(CHANNEL_spTemp);
+        verifyLinkedChannel(CHANNEL_ecoMode);
+        verifyLinkedChannel(CHANNEL_ignitions);
+        verifyLinkedChannel(CHANNEL_maintenanceIn);
+        verifyLinkedChannel(CHANNEL_cleaningIn);
+        verifyLinkedChannel(CHANNEL_consumption);
+        verifyLinkedChannel(CHANNEL_onTime);
+
+        if (!linkedChannels.isEmpty()) {
+            logger.debug("Start automatic refreshing");
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging